avocado/plugins/diff.py

```diff
@@ -0,0 +1,419 @@
+        with open(results_json, 'r') as json_file:
+            data = json.load(json_file)
+
+        return data
+
+    @staticmethod
+    def _setup_job(job_id):
+        if os.path.isdir(job_id):
+            resultsdir = os.path.expanduser(job_id)
+            job_id = ''
+        elif os.path.isfile(job_id):
+            resultsdir = os.path.dirname(os.path.expanduser(job_id))
+            job_id = ''
+        else:
+            logs_dir = settings.get_value('datadir.paths', 'logs_dir',
+                                          default=None)
+            logdir = os.path.expanduser(logs_dir)
+            resultsdir = replay.get_resultsdir(logdir, job_id)
+
+        if resultsdir is None:
+            LOG.error("Can't find job results directory for '%s' in '%s'",
+                      job_id, logdir)
+            sys.exit(exit_codes.AVOCADO_FAIL)
+
+        sourcejob = replay.get_id(os.path.join(resultsdir, 'id'), job_id)
+        if sourcejob is None:
+            LOG.error("Can't find matching job id '%s' in '%s' directory.",
+                      job_id, resultsdir)
+            sys.exit(exit_codes.AVOCADO_FAIL)
+
+        return resultsdir, sourcejob
+
+    @staticmethod
+    def _get_command_line(resultsdir):
+        command_line = replay.retrieve_cmdline(resultsdir)
+        if command_line is not None:
+            return '%s\n' % ' '.join(command_line)
+
+        return 'Not found\n'
+
+    @staticmethod
+    def _get_variants(resultsdir):
+        results = []
+        mux = replay.retrieve_mux(resultsdir)
+        if mux is not None:
+            env = set()
+            for (index, tpl) in enumerate(mux.variants):
+                paths = ', '.join([x.path for x in tpl])
+                results.append('Variant %s: %s\n' % (index + 1, paths))
+                for node in tpl:
+                    for key, value in node.environment.iteritems():
+                        origin = node.environment_origin[key].path
+                        env.add(("%s:%s" % (origin, key), str(value)))
+                if not env:
+                    continue
+                fmt = '    %%-%ds => %%s\n' % max([len(_[0]) for _ in env])
+                for record in sorted(env):
+                    results.append(fmt % record)
+        else:
+            results.append('Not found\n')
+
+        return results
+
+    @staticmethod
+    def _get_config(resultsdir):
+        config_file = os.path.join(resultsdir, 'replay', 'config')
+        try:
+            with open(config_file, 'r') as conf:
+                return conf.readlines()
+        except IOError:
+            return ['Not found\n']
+
+    @staticmethod
+    def _get_sysinfo(resultsdir, pre_post):
+        sysinfo_dir = os.path.join(resultsdir, 'sysinfo', pre_post)
+        sysinfo = []
+        for path, _, files in os.walk(sysinfo_dir):
+            for name in sorted(files):
+                name_header = '** %s **\n' % name
```

Comment:
This version is a bit too compressed:

```
# SYSINFO PRE
** brctl_show **
bridge name bridge id       STP enabled interfaces
virbr0      8000.525400910f38   yes     virbr0-nic
** cmdline **
BOOT_IMAGE=/vmlinuz-4.6.4-201.fc23.x86_64 root=/dev/mapper/luks-c5fd3fc3-933b-4a44-8f03-baa604657e25 ro rd.luks.uuid=luks-c5fd3fc3-933b-4a44-8f03-baa604657e25 rd.lvm.lv=fedora_dhcp-27-34/02 rd.lvm.lv=fedora_dhcp-27-34/00 rd.luks.uuid=luks-581d6aac-9e7b-4cee-bb89-f583c753f560 LANG=cs_CZ.UTF-8 pcie_aspm=force acpi_osi=Linux video.use_native_backlight=1 i915.enable_rc6=7 i915.i915_enable_rc6=7
** cpuinfo **
processor   : 0
vendor_id   : GenuineIntel
```

Let's find something in between (force update is welcome)

I'd go with
```
diff --git a/avocado/plugins/diff.py b/avocado/plugins/diff.py
index 94c3bfc..655af40 100644
--- a/avocado/plugins/diff.py
+++ b/avocado/plugins/diff.py
@@ -400,11 +400,12 @@ class Diff(CLICmd):
         sysinfo = []
         for path, _, files in os.walk(sysinfo_dir):
             for name in sorted(files):
-                name_header = '** %s **\n' % name
+                name_header = '\n** %s **\n' % name
                 sysinfo.append(name_header)
                 with open(os.path.join(path, name), 'r') as sysinfo_file:
                     sysinfo.extend(sysinfo_file.readlines())
-
+        if sysinfo:
+            sysinfo[0] = sysinfo[0][1:]
         return sysinfo
 
     def _cdiff(self, diff):
```